rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/selection/topology/TopologyManager.java

```diff
@@ -141,24 +146,33 @@ public synchronized void addNode(Node node) {
         synchronized (node.getNodeInformation().getURL().intern()) {
             // unknown host => start pinging process
             NodeSet toPing = new NodeSet();
+            HashMap<InetAddress, Long> hostsTopology = new HashMap<>();
+
             synchronized (topology) {
                 // adding one node from each host
                 for (InetAddress h : nodesOnHost.keySet()) {
                     // always have at least one node on each host
-                    toPing.add(nodesOnHost.get(h).iterator().next());
+                    if (nodesOnHost.get(h) != null && !nodesOnHost.get(h).isEmpty()) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1155)